static/app/components/arithmeticBuilder/grammar.pegjs

```diff
@@ -0,0 +1,92 @@
+{
+  const {tc, term} = options;
+}
+
+tokens = token*
+
+token
+  = spaces token:(paren / op / func / free_text) spaces {
+    return token;
+  }
+
+func
+  = func:name "(" spaces attr:attr spaces ")" {
+    return tc.tokenFunction(func, attr, location());
+  }
+
+attr = typed_attr / untyped_attr
+
+typed_attr
+  = "tags[" name:name "," spaces type:type_name "]" {
+    return tc.tokenAttribute(name, type, location());
+  }
+
+untyped_attr
+  = name:name {
+    return tc.tokenAttribute(name, undefined, location());
+  }
+
+free_text = typed_attr_string / untyped_attr_string_or_func_string / string
+
+string
+  = [^ ]+ {
+    return tc.tokenFreeText(text(), location());
+  }
+
+untyped_attr_string_or_func_string
+  = name ("(" (spaces (typed_attr_string / untyped_attr_string) (spaces ")")?)?)? {
+    return tc.tokenFreeText(text(), location());
+  }
+
+typed_attr_string
+  = "t" ("a" ("g" ("s" ("[" (name ("," (spaces type_name ("]")?)?)?)?)?)?)?)? {
+    return tc.tokenFreeText(text(), location());
+  }
+
+untyped_attr_string
+  = name:name {
+    return text();
+  }
+
+name
+  = [^()\t\n, \"]+ {
+    return text();
+  }
+
+type_name
+  = [a-z]+ {
+    return text();
+  }
```

Comment:
does it make sense to make is explicitly the attr types that are supported?

Reply:
Intentionally keeping this loosely typed to allow for creation of tokens even if it's invalid while they're typing it. I can look into making it more strict and see what happens later.